central/graphql/resolvers/node_vulnerabilities.go

```diff
@@ -80,8 +82,366 @@ func (resolver *Resolver) NodeVulnCounter(ctx context.Context, args RawQuery) (*
+	scope, hasScope := scoped.GetScope(resolver.ctx)
+	if !hasScope {
+		return "", nil
+	}
+	if scope.Level != v1.SearchCategory_NODE_COMPONENTS {
+		return "", nil
+	}
+	edgeID := edges.EdgeID{ParentID: scope.ID, ChildID: resolver.data.GetId()}.ToString()
+	edge, found, err := resolver.root.NodeComponentCVEEdgeDataStore.Get(resolver.ctx, edgeID)
+	if err != nil || !found {
+		return "", err
+	}
+	return edge.GetFixedBy(), nil
+}
+
+// ID of the node CVE
+func (resolver *nodeCVEResolver) ID(ctx context.Context) graphql.ID {
+	value := resolver.data.GetId()
+	return graphql.ID(value)
+}
+
+// IsFixable returns whether node CVE is fixable by any component
+func (resolver *nodeCVEResolver) IsFixable(ctx context.Context, args RawQuery) (bool, error) {
+	// TODO : Why do we remove this field query and then add it back again in addScopeContextOrNodeCVEQuery ?
+	q, err := args.AsV1QueryOrEmpty(search.ExcludeFieldLabel(search.CVE))
+	if err != nil {
+		return false, err
+	}
+
+	ctx, query := resolver.addScopeContextOrNodeCVEQuery(q)
+	query = search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery())
+	count, err := resolver.root.NodeComponentCVEEdgeDataStore.Count(ctx, query)
+	if err != nil {
+		return false, err
+	}
+	return count != 0, nil
+}
+
+func (resolver *nodeCVEResolver) addScopeContextOrNodeCVEQuery(query *v1.Query) (context.Context, *v1.Query) {
+	ctx := resolver.ctx
+	scope, ok := scoped.GetScope(ctx)
+	if !ok {
+		return resolver.withNodeVulnerabilityScope(ctx), query
+	}
+	// If the scope is not set to node vulnerabilities then
+	// we need to add a query to scope the search to the current nodeCVE
+	if scope.Level != v1.SearchCategory_NODE_VULNERABILITIES {
+		return ctx, search.ConjunctionQuery(query, resolver.getNodeCVEQuery())
+	}
+
+	return ctx, query
+}
+
+func (resolver *nodeCVEResolver) addScopeContextOrNodeCVERawQuery(query string) (context.Context, string) {
+	ctx := resolver.ctx
+	scope, ok := scoped.GetScope(ctx)
+	if !ok {
+		return resolver.withNodeVulnerabilityScope(ctx), query
+	}
+	// If the scope is not set to node vulnerabilities then
+	// we need to add a query to scope the search to the current nodeCVE
+	if scope.Level != v1.SearchCategory_NODE_VULNERABILITIES {
+		return ctx, search.AddRawQueriesAsConjunction(query, resolver.getNodeCVERawQuery())
+	}
+
+	return ctx, query
+}
+
+func (resolver *nodeCVEResolver) getNodeCVEQuery() *v1.Query {
+	return search.NewQueryBuilder().AddExactMatches(search.CVE, resolver.data.GetId()).ProtoQuery()
+}
+
+func (resolver *nodeCVEResolver) getNodeCVERawQuery() string {
+	return search.NewQueryBuilder().AddExactMatches(search.CVE, resolver.data.GetId()).Query()
+}
+
+// LastModified is the time this node CVE was last modified in the system
+func (resolver *nodeCVEResolver) LastModified(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetCveBaseInfo().GetLastModified()
+	return timestamp(value)
```

Comment:
I noticed this in a couple of other places as well, but I don't think you need to cast here, you should be able to just return the type you get. Also I think we need to check whether the value returned from the base info function is null before we call on it

Reply:
nit: can be one lined